### PR TITLE
[config] Update build/test config experience; add signal to support partner developers

### PR DESCRIPTION
**Pull request by arjkesh**
*Issue #, if available:*

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON Checklist
* When creating a PR:
- [ ] I've modified `src/config/build_config.py` in my PR branch by setting `ENABLE_EI_MODE = True` or `ENABLE_NEURON_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above
#### Benchmark Checklist
* When creating a PR:
- [ ] I've modified `src/config/test_config.py` in my PR branch by setting `ENABLE_BENCHMARK_DEV_MODE = True`
* When PR is reviewed and ready to be merged:
- [ ] I've reverted the code change on the config file mentioned above

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [ ] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Add toml file for all developer configs. Add script to parse the toml and send signal for partner developers. Add tests to ensure defaults of config toml are appropriate.

*Tests run:*
Testing in this PR, added sanity tests

*Additional Context*
Partner teams will be able to identify themselves in PRs in order to get access to logs

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: dlc_developer_config.toml
Comment (arjkesh) [inline dlc_developer_config.toml:3]: temporary change

File: dlc_developer_config.toml
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:3]: is this a boolean value or string?

File: parse_partner_developers.py
Comment (tejaschumbalkar) [inline parse_partner_developers.py:39]: why is this needed? just for logging?

File: dlc_developer_config.toml
Comment (tejaschumbalkar) [reply dlc_developer_config.toml:3]: What is the significance for this field?

File: src/config/build_config.py
Comment (tejaschumbalkar) [inline src/config/build_config.py:5]: Eventually we are adding a wrapper for the temp configs?

**Comment by tejaschumbalkar** [comment]
Also, we would be reverting the modifications to `dlc_developer_config.toml` once the a PR is merge ready?

File: parse_partner_developers.py
Comment (arjkesh) [reply parse_partner_developers.py:39]: This script signals in our logs if there is a partner developer writing code

File: dlc_developer_config.toml
Comment (arjkesh) [reply dlc_developer_config.toml:3]: Good point, probably a more sensible default is ""

This is basically to signal that a build is associated with a given partner

File: src/config/build_config.py
Comment (arjkesh) [reply src/config/build_config.py:5]: eventually, yeah we can remove this layer of abstraction

**Comment by arjkesh** [comment]
> Also, we would be reverting the modifications to `dlc_developer_config.toml` once the a PR is merge ready?

yes, there is also a sanity test that will fail if the defaults are not restored. any modifications must be reverted before merge

**Comment by tejaschumbalkar** [comment]
LGTM. Will approve once the temp configs are reverted

File: dlc_developer_config.toml
Comment (tejaschumbalkar) [inline dlc_developer_config.toml:3]: Should we add the expected values for `partner_developer` in the comments?

Comment (arjkesh) [reply dlc_developer_config.toml:3]: added

**Review by tejaschumbalkar — APPROVED** [review]

**Pull request merged**
